src/main/java/io/appium/java_client/android/nativekey/KeyEvent.java

```diff
@@ -83,8 +84,8 @@ public KeyEvent withFlag(KeyEventFlag keyEventFlag) {
      */
     public Map<String, Object> build() {
         var map = new HashMap<String, Object>();
-        ofNullable(this.keyCode).map(x -> map.put("keycode", x)).orElseThrow(() -> new IllegalStateException(
```

Comment:
@valfirst Could there be other places in the source like this one? I assume the issue may have been added during the recent refactoring

Reply:
yes, this is a regression issue, I've found 2 more similar occurrences, I'll open PR soon

Reply:
thanks. Lets address them all and publish a patch asap

Reply:
@mykola-mokhnach I believe I fixed then in my last commit

Reply:
I've already addressed `AndroidGeoLocation` issues here: #2146